### PR TITLE
feat(angular, react, vue): add support for autoMountComponent

### PR DESCRIPTION
**Pull request by liamdebeasi**
Docs: https://github.com/ionic-team/ionic-docs/pull/2432

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: Internal ticket.

## What is the new behavior?

- Adds the ability to have inline modals and popovers automatically mount their inner contents on creation. The contents will only be unmounted when the parent modal/popover is unmounted. Re-mounting the inner modal/popover does not preserve the inner content state.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by amandaejohnston** [comment]
Maybe `keepContentsMounted`? It's similar to the existing `keepMounted` API, but it makes it more clear what exactly we're mounting.

File: angular/test/test-app/e2e/src/overlay-auto-mount.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+describe('overlays - keepContentsMounted', () => {
```
Comment (sean-perkins) [inline angular/test/test-app/e2e/src/overlay-auto-mount.spec.ts:1]: ~ May want to rename this file to match the updated property name.

File: angular/test/test-app/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ const routes: Routes = [
   { path: 'modals', component: ModalComponent },
   { path: 'modal-inline', loadChildren: () => import('./modal-inline').then(m => m.ModalInlineModule) },
   { path: 'view-child', component: ViewChildComponent },
+  { path: 'overlay-auto-mount', loadChildren: () => import('./overlay-auto-mount').then(m => m.OverlayAutoMountModule) },
```
Comment (sean-perkins) [inline angular/test/test-app/src/app/app-routing.module.ts:34]: ~ May want to rename path to match the prop name.

File: angular/test/test-app/src/app/overlay-auto-mount/overlay-auto-mount.component.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Component } from "@angular/core";
+
+/**
+ * Validates that inline modals correctly mount
+ * inner components when keepContentsMounted is
+ * enabled.
+ */
+@Component({
+  selector: 'app-overlay-auto-mount',
```
Comment (sean-perkins) [inline angular/test/test-app/src/app/overlay-auto-mount/overlay-auto-mount.component.ts:9]: ~ File name, selector and class name should have the same naming conventions. 

I would recommend the class be called: `OverlayKeepContentsMountedComponent` to align with Angular style guides for component naming.

File: packages/react/test-app/cypress/integration/overlay-components/AutoMount.spec.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+describe('keepContentsMounted', () => {
```
Comment (sean-perkins) [inline packages/react/test-app/cypress/integration/overlay-components/AutoMount.spec.ts:1]: ~ Should we rename this spec file to match the prop name?

**Review by sean-perkins — COMMENTED** [review]
Minor suggestions. Unsure how much of the language should be around the prop name vs. the behavior of the feature. I do not have any strong opinions on either.

**Review by amandaejohnston — APPROVED** [review]

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**